### PR TITLE
feat(tls): support ca type issuer and v1alpha* version cert-manager api

### PR DESCRIPTION
**Pull request by ericsyh**
Fixes #<xyz>

### Motivation

The current chart only supports enable tls with self-signed issuer which is not possible for production users. This PR supports [ca type issuser](https://cert-manager.io/docs/configuration/ca/) which allows users to bound a root certificate (in `Secret`) to issue required tls certificates for pulsar components. 

### Modifications

- support the ca type issuer in the `certs.type` and add the `certs.issuers.ca.secretName` field for the root certificate reference. 
- support cert-manager v1alpha[1-3] API versions. 

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### CODE REVIEW

File: charts/pulsar/templates/tls-certs-internal.yaml [hunk 1]
```diff
@@ -32,27 +32,38 @@ spec:
...
+  keyAlgorithm: {{ .Values.tls.common.keyAlgorithm }}
+  keyEncoding: {{ .Values.tls.common.keyEncoding }}
+  {{- end }}
   usages:
     - server auth
     - client auth
   # At least one of a DNS Name, USI SAN, or IP address is required.
   dnsNames:
-    -  "*.{{ template "pulsar.fullname" . }}-{{ .Values.proxy.component }}.{{ template "pulsar.namespace" . }}.svc.{{ .Values.clusterDomain }}"
-    -  "{{ template "pulsar.fullname" . }}-{{ .Values.proxy.component }}"
```
Comment (lhotari) [inline charts/pulsar/templates/tls-certs-internal.yaml:52]: Is this removal a breaking change? Does this require release note instructions users that upgrade? Would it be possible to retain compatibility?

**Review by lhotari — COMMENTED** [review]
Good work! I added a question regarding backwards compatibility.

File: charts/pulsar/templates/tls-certs-internal.yaml (hunk 1 above)
Comment (ericsyh) [reply charts/pulsar/templates/tls-certs-internal.yaml:52]: Nope, it's not a breaking change cause i just move the position to make the format of [proxy certificate dnsNames](https://github.com/apache/pulsar-helm-chart/blob/master/charts/pulsar/templates/tls-certs-internal.yaml#L50) and [broker certificate dnsNames](https://github.com/apache/pulsar-helm-chart/blob/master/charts/pulsar/templates/tls-certs-internal.yaml#L95) to be consistent. It doesn't change the logic and technical this change is optional.

**Review by ericsyh — COMMENTED** [review]
replay the comment.

**Review by ciiiii — APPROVED** [review]
LGTM

**Comment by lhotari** [comment]
@ericsyh [linting fails in `charts/pulsar/values.yaml`](https://github.com/apache/pulsar-helm-chart/actions/runs/12345620973/job/34583492482?pr=561#step:8:32)
```
  Error: 236:21 [trailing-spaces] trailing spaces
  Error: 245:14 [trailing-spaces] trailing spaces
  Error: 314:105 [trailing-spaces] trailing spaces
```

**Comment by ericsyh** [comment]
> @ericsyh [linting fails in `charts/pulsar/values.yaml`](https://github.com/apache/pulsar-helm-chart/actions/runs/12345620973/job/34583492482?pr=561#step:8:32)
> 
> ```
>   Error: 236:21 [trailing-spaces] trailing spaces
>   Error: 245:14 [trailing-spaces] trailing spaces
>   Error: 314:105 [trailing-spaces] trailing spaces
> ```

@lhotari fixed in https://github.com/apache/pulsar-helm-chart/pull/561/commits/700a437c9ea6aaa847d423248601a16c7b40b4a2, could you trigger the CI again?

**Pull request merged**